RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h

```diff
@@ -151,27 +141,22 @@ void HcalDeterministicFit::apply(const CaloSamples & cs, const std::vector<int>
         double invG = invGpar[0]+invGpar[1]*std::sqrt(2*std::log(invGpar[2]/ratio));
         float iG=0;
         getLandauFrac(-invG,-invG+tsWidth,iG);
-        ch4=(corrCharge[4]-ch3*n3)/(iG);
-        ch5=negThresh[0];
-        tsShift4=invG;
+        if (iG != 0 ) {
+	  ch4=(corrCharge[4]-ch3*n3)/(iG);
+	  ch5=negThresh[0];
```

Comment:
@cvuosalo if your students finds other code lines to cleanup in these and other packages will be just great to have them integrated (of course another PR)